projectq/ops/_basics.py

```diff
@@ -295,23 +295,29 @@ def get_merged(self, other):
 
     def __eq__(self, other):
         """ Return True if same class and same rotation angle. """
-        tolerance = 1.e-12
-        if isinstance(other, self.__class__):
-            difference = abs(self._angle - other._angle)
```

Comment:
The precondition of this class is that the angle is in the interval [0, 4*pi). Your construction made it "safer" as it still worked if that was not the case. We can easily add this here by
`difference = abs(self._angle - other._angle) % (4 * math.pi)`